cmd/fs-v1.go

```diff
@@ -926,8 +926,8 @@ func (fs *FSObjects) putObject(ctx context.Context, bucket string, object string
 	// Entire object was written to the temp location, now it's safe to rename it to the actual location.
 	fsNSObjPath := pathJoin(fs.fsPath, bucket, object)
 	// Deny if WORM is enabled
-	if globalWORMEnabled {
-		if _, err = fsStatFile(ctx, fsNSObjPath); err == nil {
+	if retention, isWORMBucket := isWORMEnabled(bucket); isWORMBucket {
```

Comment:
This is duplicated soo much. Would it be possible to have this on the fs object?

I am sure sure how it could be abstracted, but ideally we should wrap the logic, so the code is simplified to:
```
	if !fs.CanOverwrite(bucket, object) {
		return ObjectInfo{}, ObjectAlreadyExists{Bucket: bucket, Object: object}
	}
```

Reply:
Yes. This can be done. When we move WORM logic from handlers to object layer, this can be done. I am just keeping the logic as is to work exactly same as existing global WORM mode.